lnd_test.go

```diff
@@ -99,12 +100,107 @@ func testBasicChannelFunding(net *networkHarness, t *testing.T) {
+	ctxb := context.Background()
+
+	openChannel := func(alice lnrpc.LightningClient, bob lnrpc.LightningClient, amount float64) *lnrpc.ChannelPoint {
+		// First establish a channel ween with a capacity of 0.5 BTC between
+		// Alice and Bob.
+		chanAmt := btcutil.Amount(amount)
+		chanOpenUpdate, err := net.OpenChannel(ctxb, alice, bob, chanAmt, 1)
+		if err != nil {
+			t.Fatalf("unable to open channel: %v", err)
+		}
+
+		// Mine a block, then wait for Alice's node to notify us that the
+		// channel has been opened. The funding transaction should be found
+		// within the newly mined block.
+		blockHash, err := net.Miner.Node.Generate(1)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+		block, err := net.Miner.Node.GetBlock(blockHash[0])
+		if err != nil {
+			t.Fatalf("unable to get block: %v", err)
+		}
+		fundingChanPoint, err := net.WaitForChannelOpen(chanOpenUpdate)
+		if err != nil {
+			t.Fatalf("error while waiting for channeel open: %v", err)
+		}
+		fundingTxID, err := wire.NewShaHash(fundingChanPoint.FundingTxid)
+		if err != nil {
+			t.Fatalf("unable to create sha hash: %v", err)
+		}
+		assertTxInBlock(block, fundingTxID, t)
+
+		// The channel should be listed in the peer information returned by
+		// both peers.
+		chanPoint := wire.OutPoint{
+			Hash:  *fundingTxID,
+			Index: fundingChanPoint.OutputIndex,
+		}
+		err = net.AssertChannelExists(ctxb, net.AliceClient, &chanPoint)
+		if err != nil {
+			t.Fatalf("unable to assert channel existence: %v", err)
+		}
+
+		return fundingChanPoint
+	}
+
+	closeChannel := func(node lnrpc.LightningClient, fundingChanPoint *lnrpc.ChannelPoint) {
+		// Initiate a close
+		closeUpdates, err := net.CloseChannel(ctxb, node, fundingChanPoint, false)
+		if err != nil {
+			t.Fatalf("unable to clsoe channel: %v", err)
+		}
+
+		// Finally, generate a single block, wait for the final close status
+		// update, then ensure that the closing transaction was included in the
+		// block.
+		blockHash, err := net.Miner.Node.Generate(1)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+		block, err := net.Miner.Node.GetBlock(blockHash[0])
+		if err != nil {
+			t.Fatalf("unable to get block: %v", err)
+		}
+
+		closingTxid, err := net.WaitForChannelClose(closeUpdates)
+		if err != nil {
+			t.Fatalf("error while waiting for channel close: %v", err)
+		}
+		assertTxInBlock(block, closingTxid, t)
+
+	}
+
+	checkChannelBalance := func (node lnrpc.LightningClient, amount float64) {
+		response, err := node.ChannelBalance(ctxb, &lnrpc.ChannelBalanceRequest{})
+		if err != nil {
+			t.Fatalf("unable to get channel balance: %v", err)
+		}
+
+		if response.Balance != amount {
+			t.Fatalf("channel balance wrong: %v != %v", response.Balance, amount)
+		}
+	}
+
+	amount := btcutil.SatoshiPerBitcoin / 2
+	chanPoint := openChannel(net.AliceClient, net.BobClient, amount)
+
+	checkChannelBalance(net.BobClient, amount)
```

Comment:
Because we wait only for `Alice` `OpenStatusUpdate_ChanOpen` updates  (as far us I know in case of `Alice` channel creation initiation we could get only `Alice` channel updates), might happen that channel on `Alice` side was added in `activeChannels` list (`capacityBalance` works properly)  but on `Bob` side we haven't done that yet. So I may add `time.Sleep` or maybe there is better way?